dev/integration_tests/android_semantics_testing/test_driver/main_test.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' as io;
+
+import 'package:android_semantics_testing/test_constants.dart';
+import 'package:android_semantics_testing/android_semantics_testing.dart';
+
+import 'package:test/test.dart' hide TypeMatcher, isInstanceOf;
+import 'package:flutter_driver/flutter_driver.dart';
+
+void main() {
+  group('AccessibilityBridge', () {
+    FlutterDriver driver;
+    Future<AndroidSemanticsNode> getSemantics(SerializableFinder finder) async {
+      final int id = await driver.getSemanticsId(finder);
+      final String data = await driver.requestData('getSemanticsNode#$id');
+      return new AndroidSemanticsNode.deserialize(data);
+    }
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+      // Say the magic words..
+      final io.Process run = await io.Process.start('adb', const <String>[
+        'shell',
+        'settings',
+        'put',
+        'secure',
+        'enabled_accessibility_services',
+        'com.google.android.marvin.talkback/com.google.android.marvin.talkback.TalkBackService',
+      ]);
+      await run.exitCode;
+    });
+
+    tearDownAll(() async {
+      // ... And turn it off again
+      final io.Process run = await io.Process.start('adb', const <String>[
```

Comment:
I think the agent code should do the clean-up (perhaps _in addition_ to the clean-up here). Otherwise, this is pretty much guaranteed to leak into other tests. The proper place to add it is somewhere in the CI loop, perhaps next to the place where we turn off the screen: https://github.com/flutter/cocoon/blob/master/agent/lib/src/commands/ci.dart#L120

Reply:
That would require refactoring and updating the cocoon agent code to dart 2 as well...

You're right though, we'll need to expose some sort of AndroidAccessibilityTest config to handle this.

(_But_  leaving talkback on shouldn't negatively effect testing, since all of the driver gestures bypass the talkback gestures, just saying)

Reply:
cc @goderbauer , @gspencergoog do you guys agree with yegor that we need to do this in coccon agent? tearDownAll is always called so I believe the chance of it failing to execute are minimal - or rather the agent process invoking it has the same risks.

Reply:
Leaving it up to @goderbauer and @gspencergoog (and perhaps @cbracken) to make a final call, . If I was among the ones maintaining the devicelab right now I'd continue insisting that the agents do the clean-up :smile: I built the devicelab with the philosophy that flutter/flutter repository cannot be trusted, and where possible (and practical) did all the clean-up in the agent code, which doesn't change often. For example, the agents are currently reponsible for managing screen on/off state, cleaning up dangling processes, and smoke-testing the USB connections. I think this falls under the same category.

Reply:
I would agree with Yegor that the clean-up should also happen in the agent. Is turning it off a no-op if it is already off? In that case we could turn it off before we run a test to ensure that this doesn't leak across tests even if the agent or test was killed mid-execution?

@jonahwilliams Why is this dependent on moving the agents to Dart 2?

Reply:
Actually we might not need to move to dart 2 I think I was confused a bit with something else. Turning it off is a no-op, so we can run that as a part of clean up. In that case, the test would still be required to set up TalkBack and could tear it down too, but there would be way less risk of leaks. @yjbanov does that sound good to you?

Reply:
:+1:

Reply:
I've updated linux1/2 ... just a bunch of macs to go

Reply:
macs updated